components/domains/registerV2.tsx

```diff
@@ -0,0 +1,363 @@
+import styles from "../../styles/components/registerV2.module.css";
+import TextField from "../UI/textField";
+// import UsForm from "./usForm";
+import { Divider } from "@mui/material";
+import RegisterCheckboxes from "./registerCheckboxes";
+import NumberTextField from "../UI/numberTextField";
+import RegisterSummary from "./registerSummary";
+import salesTax from "sales-tax";
+import Wallets from "../UI/wallets";
+import registerCalls from "../../utils/registerCalls";
+
+type RegisterV2Props = {
+  domain: string;
+};
+
+const RegisterV2: FunctionComponent<RegisterV2Props> = ({ domain }) => {
+  const maxYearsToRegister = 25;
+  const [targetAddress, setTargetAddress] = useState<string>("");
+  const [email, setEmail] = useState<string>("");
+  const [emailError, setEmailError] = useState<boolean>(false);
+  const [isUsResident, setIsUsResident] = useState<boolean>(false);
+  const [usState, setUsState] = useState<string>("DE");
+  const [salesTaxRate, setSalesTaxRate] = useState<number>(0);
+  const [duration, setDuration] = useState<number>(1);
+  const [tokenId, setTokenId] = useState<number>(0);
+  const [callData, setCallData] = useState<Call[]>([]);
+  const [price, setPrice] = useState<string>("0");
+  const [balance, setBalance] = useState<string>("0");
+  const [invalidBalance, setInvalidBalance] = useState<boolean>(false);
+  const { contract } = usePricingContract();
+  const { contract: etherContract } = useEtherContract();
+  const [isTxModalOpen, setIsTxModalOpen] = useState(false);
+  const encodedDomain = utils
+    .encodeDomain(domain)
+    .map((element) => element.toString())[0];
+  const [termsBox, setTermsBox] = useState<boolean>(true);
+  const [renewalBox, setRenewalBox] = useState<boolean>(true);
+  const [walletModalOpen, setWalletModalOpen] = useState<boolean>(false);
+  const [sponsor, setSponsor] = useState<string>("0");
+
+  const { data: priceData, error: priceError } = useContractRead({
+    address: contract?.address as string,
+    abi: contract?.abi as Abi,
+    functionName: "compute_buy_price",
+    args: [encodedDomain, duration * 365],
+  });
+
+  const { account, address } = useAccount();
+  const { data: userBalanceData, error: userBalanceDataError } =
+    useContractRead({
+      address: etherContract?.address as string,
+      abi: etherContract?.abi as Abi,
+      functionName: "balanceOf",
+      args: [address],
+    });
+  const { writeAsync: execute, data: registerData } = useContractWrite({
+    calls: renewalBox
+      ? callData.concat(registerCalls.renewal(encodedDomain, price))
+      : callData,
+  });
+  const hasMainDomain = !useDisplayName(address ?? "", false).startsWith("0x");
+  const [domainsMinting, setDomainsMinting] = useState<Map<string, boolean>>(
+    new Map()
+  );
+  const { addTransaction } = useTransactionManager();
+
+  useEffect(() => {
+    if (priceError || !priceData) setPrice("0");
+    else {
+      const high = priceData?.["price"].high << BigInt(128);
```

Comment:
Are you sure << is supported by BigInt?

Reply:
Yes it was here before I just copy pasted